### PR TITLE
Add config function to set max number of connections for http finder server

### LINKED ISSUE #632: Maximum number of finder connections should be configurable in config file
There is an existing option [here](https://github.com/filecoin-project/storetheindex/blob/main/server/finder/http/options.go#L31), but needs a function to set the option.

### PR DESCRIPTION
**Pull request by arnabguin**
Default values can be overridden programatically through the config function or by altering the Finder section of the config file.

Fixes #632

### CODE REVIEW

File: config/finder.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func NewFinder() Finder {
+	return Finder{
+		ApiWriteTimeout: Duration(30 * time.Second),
+		ApiReadTimeout:  Duration(30 * time.Second),
+		MaxConnections:  8_000,
+	}
+}
+
+// populateUnset replaces zero-values in the config with default values.
+func (f *Finder) populateUnset() {
```
[the remark this answers is not in the document]
Comment (arnabguin) [reply config/finder.go:25]: fixed

**Review by gammazero — APPROVED** [review]
LGTM

File: command/daemon.go [hunk 2]
```diff
@@ -157,8 +157,11 @@ func daemonCommand(cctx *cli.Context) error {
 		if err != nil {
 			return err
 		}
-		finderOpt := httpfinderserver.WithHomepage(cfg.Addresses.FinderWebpage)
-		finderSvr, err = httpfinderserver.New(finderAddr.String(), indexerCore, reg, finderOpt)
+		finderOpts := []httpfinderserver.ServerOption{
+			httpfinderserver.WithHomepage(cfg.Addresses.FinderWebpage),
+			httpfinderserver.MaxConnections(cfg.Finder.MaxConnections),
+		}
+		finderSvr, err = httpfinderserver.New(finderAddr.String(), indexerCore, reg, finderOpts...)
```
Comment (gammazero) [inline command/daemon.go:164]: Need to set read and write timeouts.
```suggestion
		finderSvr, err = httpfinderserver.New(finderAddr.String(), indexerCore, reg,
			httpfinderserver.WithHomepage(cfg.Addresses.FinderWebpage),
			httpfinderserver.MaxConnections(cfg.Finder.MaxConnections),
			httpfinderserver.ReadTimeout(time.Duration(cfg.Finder.ApiReadTimeout)),
			httpfinderserver.WriteTimeout(time.Duration(cfg.Finder.ApiWriteTimeout)),
		)
```

Also, I think that `cfg.Addresses.FinderWebpage` should be moved to `cfg.Finder.Webpage`.

Comment (arnabguin) [reply command/daemon.go]: Thanks @willscott @gammazero for the review. 
Updated with read/write timeouts.

Will look at the option movement.

**Review by willscott — APPROVED** [review]

**Pull request merged**